### PR TITLE
UIMARCAUTH-70: Results list > Clicking on Result should highlight the selected row

### PR DESCRIPTION
**Pull request by mariiaschv**
## Purpose
Highlight the selected row when click on Result in Results list.

## Screenshots
epED6ebAOy

## Issues
[UIMARCAUTH-70](https://issues.folio.org/browse/UIMARCAUTH-70)

### CODE REVIEW

File: src/RecordRowContext.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { createContext } from 'react';
+
+export const RecordRowContext = createContext();
```
Comment (BogdanDenis) [inline src/RecordRowContext.js:3]: I like the approach with context, just would like to suggest a few changes:
Name could be a bit more specific, maybe `SelectedAuthorityRecordContext`.

Also, it's a common practice to create two exports for one context file:
1. created context
2. context provider. You can create here a component `SelectedAuthorityRecordContextProvider` , move `useState` from `src/index` into this component and return `<SelectedAuthorityRecordContext.Provider>`

Also, please create `src/context` folder and move this file there

File: src/components/SearchResultsList/SearchResultsList.test.js [hunk 2]
```diff
@@ -75,6 +81,18 @@ describe('Given SearchResultsList', () => {
...
 
+  describe('when click on a row', () => {
+    it('should handle setRecordRowContext', () => {
+      const { getAllByRole } = renderSearchResultsList();
+
+      const [rowLink] = getAllByRole('link', { name: 'Twain, Mark' });
+
+      fireEvent.click(rowLink);
+
+      expect(mockSetRecordRowContext).toHaveBeenCalled();
```
Comment (BogdanDenis) [inline src/components/SearchResultsList/SearchResultsList.test.js:92]: Please also check that it was called with correct object

File: src/views/AuthoritiesSearch/AuthoritiesSearch.test.js [hunk 3]
```diff
@@ -151,6 +155,23 @@ describe('Given AuthoritiesSearch', () => {
...
+      const textarea = getByTestId('search-textarea');
+      const searchButton = getByTestId('submit-authorities-search');
+
+      fireEvent.change(textarea, { target: { value: 'test search' } });
+
+      expect(textarea.value).toBe('test search');
+
+      fireEvent.click(searchButton);
+
+      expect(mockSetRecordRowContext).toHaveBeenCalled();
```
Comment (BogdanDenis) [inline src/views/AuthoritiesSearch/AuthoritiesSearch.test.js:171]: Also check correct object here

File: src/views/AuthoritiesSearch/AuthoritiesSearch.test.js [hunk 4]
```diff
@@ -276,6 +297,24 @@ describe('Given AuthoritiesSearch', () => {
...
+      const textarea = getByTestId('search-textarea');
+      const resetAllButton = getByRole('button', { name: 'stripes-smart-components.resetAll' });
+
+      fireEvent.change(textarea, { target: { value: 'test search' } });
+
+      expect(textarea.value).toBe('test search');
+
+      fireEvent.click(resetAllButton);
+
+      expect(mockSetRecordRowContext).toHaveBeenCalled();
```
Comment (BogdanDenis) [inline src/views/AuthoritiesSearch/AuthoritiesSearch.test.js:316]: And here

**Review by BogdanDenis — CHANGES_REQUESTED** [review]

File: src/RecordRowContext.js (hunk 1 above)
Comment (mariiaschv) [reply src/RecordRowContext.js]: Thank you for this note! Fixed now

**Review by Svitlana-Korol — APPROVED** [review]

**Pull request merged**